### PR TITLE
Feature/clean tests kcoin

### PR DESCRIPTION
**Pull request by jmatosp**
Adding tests for Voting System
Removing broken tests

### CODE REVIEW

File: kcoin/validator/consensus.go [hunk 1]
```diff
@@ -45,8 +46,8 @@ func NewVotingTables(eventMux *event.TypeMux, voters types.Voters) VotingTables
 		go eventMux.Post(core.NewMajorityEvent{})
 	}
 	tables := VotingTables{}
-	tables[0] = core.NewVotingTable(types.PreVote, voters, majorityFunc)
-	tables[1] = core.NewVotingTable(types.PreCommit, voters, majorityFunc)
+	tables[0], _ = core.NewVotingTable(types.PreVote, voters, majorityFunc)
```
[the remark this answers is not in the document]
Comment (jmatosp) [reply kcoin/validator/consensus.go]: :relaxed:
yes, agree

will actually try to add that test to CI

File: kcoin/validator/consensus.go [hunk 2]
```diff
@@ -56,21 +57,18 @@ type VotingSystem struct {
...
 	votesPerRound  map[uint64]VotingTables
-	signer         types.Signer
 
 	eventMux *event.TypeMux
 }
 
 // NewVotingSystem returns a new voting system
-// @TODO (rgeraldes) - in the future replace eventMux with a subscription method
-func NewVotingSystem(eventMux *event.TypeMux, signer types.Signer, electionNumber *big.Int, voters types.Voters) *VotingSystem {
+func NewVotingSystem(eventMux *event.TypeMux, electionNumber *big.Int, voters types.Voters) *VotingSystem {
```
[the remark this answers is not in the document]
Comment (jmatosp) [reply kcoin/validator/consensus.go]: yes, I've noticed that, but not sure how the alternative works. but a ticket is good

@rgeraldes

[the remark this answers is not in the document]
Comment (rgeraldes) [reply kcoin/validator/consensus.go]: Have a look at the implementation of this method - https://github.com/kowala-tech/kcoin/blob/27a7a453de75b816ac82d6fb847035f888dbb8fc/kusd/validator/election.go#L31. I did for other issue.

**Comment by jmatosp** [comment]
@rgeraldes yeah, AddressVote is very weird. But I had a collision of the Signed Votes.
The subsequent refactor aim for moving away from a "unsigned" vote everywhere to a always "SignedVote", this will reduce the dependency on Signer in some places.

- `Vote` object should always be a valid one, functional perspective. For that we should prob have 2, one network `Vote`(data only), that should never be visible outside the network package and validated/discard the moment it enters the app

- `Vote` that is signed/valid and is used across several places. as a developer we should NOT concern about it's validity, it has entered the app so it's valid.

The only place we would need to validate a vote is when a `Voter` wants to send a vote

This `AddressVote` is going away the moment we have this logic in place, do you have a suggestion for a better interim name?

**Pull request merged**